### PR TITLE
Csp 945 admin create event date time

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.Admin.Aan.Web.UnitTests/Controllers/ManageEvent/GuestSpeakers/GuestSpeakerControllerListTests.cs [hunk 1]
```diff
@@ -137,6 +137,6 @@ public void Post_RedirectToNextPage()
         var result = actualResult.As<RedirectToRouteResult>();
         sut.ModelState.IsValid.Should().BeTrue();
 
-        result.RouteName.Should().Be(RouteNames.ManageEvent.GuestSpeakerList);
+        result.RouteName.Should().Be(RouteNames.ManageEvent.EventDateTime);
```
Comment (ManishAgarwalDfE) [inline src/SFA.DAS.Admin.Aan.Web.UnitTests/Controllers/ManageEvent/GuestSpeakers/GuestSpeakerControllerListTests.cs:140]: The test case could be named as Post_EventDateTime page as we are posting to a specific page

File: src/SFA.DAS.Admin.Aan.Web/Validators/ManageEvent/EventDateTimeViewModelValidator.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    private bool StartTimeBeforeEndTime(EventDateTimeViewModel model, int? startHour)
+    {
+        if (model.StartHour == null || model.StartMinutes == null || model.EndHour == null ||
+            model.EndMinutes == null) return true;
+
+        var startTime = (model.StartHour * 60) + model.StartMinutes;
+        var endTime = (model.EndHour * 60) + model.EndMinutes;
+        return startTime <= endTime;
```
Comment (ManishAgarwalDfE) [inline src/SFA.DAS.Admin.Aan.Web/Validators/ManageEvent/EventDateTimeViewModelValidator.cs:74]: There are 8 conditions not covered in tests;
 
    private bool StartTimeBeforeEndTime(EventDateTimeViewModel model, int? startHour)
    {
        if (model.StartHour == null || model.StartMinutes == null || model.EndHour == null ||
            model.EndMinutes == null) return true;
        var startTime = (model.StartHour * 60) + model.StartMinutes;
        var endTime = (model.EndHour * 60) + model.EndMinutes;
        return startTime <= endTime;
    }

File: src/SFA.DAS.Admin.Aan.Web/Controllers/ManageEvent/NetworkEventDateTimeController.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    public IActionResult Post(EventDateTimeViewModel submitModel)
+    {
+        var sessionModel = _sessionService.Get<EventSessionModel>();
+        sessionModel.DateOfEvent = submitModel.DateOfEvent;
+        sessionModel.StartHour = submitModel.StartHour;
+        sessionModel.StartMinutes = submitModel.StartMinutes;
+        sessionModel.EndHour = submitModel.EndHour;
+        sessionModel.EndMinutes = submitModel.EndMinutes;
+
+        var result = _validator.Validate(submitModel);
```
Comment (ManishAgarwalDfE) [inline src/SFA.DAS.Admin.Aan.Web/Controllers/ManageEvent/NetworkEventDateTimeController.cs:46]: I think the validation can be moved before storing the values in session modal.

File: src/SFA.DAS.Admin.Aan.Web.UnitTests/Controllers/ManageEvent/GuestSpeakers/GuestSpeakerControllerListTests.cs (hunk 1 above)
Comment (MarkFCain) [reply src/SFA.DAS.Admin.Aan.Web.UnitTests/Controllers/ManageEvent/GuestSpeakers/GuestSpeakerControllerListTests.cs:140]: ok

**Review by ManishAgarwalDfE — APPROVED** [review]

**Review by ManishAgarwalDfE — APPROVED** [review]

**Review by ManishAgarwalDfE — APPROVED** [review]

**Pull request merged**